experimental/casemap/src/closer.rs

```diff
@@ -0,0 +1,226 @@
+    /// assert!(set.contains('ﬃ'));
+    ///
+    /// let mut builder = CodePointInversionListBuilder::new();
+    /// let found = cm.add_string_case_closure("ss", &mut builder);
+    /// assert!(found);
+    /// let set = builder.build();
+    ///
+    /// assert!(set.contains('ß'));
+    /// assert!(set.contains('ẞ'));
+    /// ```
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    pub const fn new() -> Self {
+        Self {
+            cm: CaseMapper::new(),
+            unfold: DataPayload::from_static_ref(
+                crate::provider::Baked::SINGLETON_PROPS_CASEMAP_UNFOLD_V1,
+            ),
+        }
+    }
+
+    icu_provider::gen_any_buffer_data_constructors!(locale: skip, options: skip, error: DataError,
+    #[cfg(skip)]
+    functions: [
+        new,
+        try_new_with_any_provider,
+        try_new_with_buffer_provider,
+        try_new_unstable,
+        Self,
+    ]);
+
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::new)]
+    pub fn try_new_unstable<P>(provider: &P) -> Result<Self, DataError>
+    where
+        P: DataProvider<CaseMapV1Marker> + DataProvider<CaseMapUnfoldV1Marker> + ?Sized,
+    {
+        let cm = CaseMapper::try_new_unstable(provider)?;
+        let unfold = provider.load(Default::default())?.take_payload()?;
+        Ok(Self { cm, unfold })
+    }
+}
+
+// We use Borrow, not AsRef, since we want the blanket impl on T
+impl<CM: Borrow<CaseMapper>> CaseMapCloser<CM> {
+    icu_provider::gen_any_buffer_data_constructors!(locale: skip, casemapper: CM, error: DataError,
+    #[cfg(skip)]
+    functions: [
+        try_new_from_casemapper_unstable,
+        try_new_from_casemapper_with_any_provider,
+        try_new_from_casemapper_with_buffer_provider,
+        try_new_from_casemapper_unstable,
+        Self,
+    ]);
+
+    /// Construct this object to wrap an existing CaseMapper (or a reference to one), loading additional data as needed.
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    ///
+    /// <div class="stab unstable">⚠️ The bounds on <tt>provider</tt> may change over time, including in SemVer minor releases.</div>
+    pub fn try_new_from_casemapper_unstable<P>(
+        provider: &P,
+        casemapper: CM,
+    ) -> Result<Self, DataError>
+    where
+        P: DataProvider<CaseMapV1Marker> + DataProvider<CaseMapUnfoldV1Marker> + ?Sized,
+    {
+        let unfold = provider.load(Default::default())?.take_payload()?;
+        Ok(Self {
+            cm: casemapper,
+            unfold,
+        })
+    }
+
+    /// Adds all simple case mappings and the full case folding for `c` to `set`.
+    /// Also adds special case closure mappings.
+    ///
+    /// In other words, this adds all strings/characters that this casemaps to, as
+    /// well as all characters that may casemap to this one.
```

Comment:
Question: Does this include all case mapping operations, or only case folding?

Reply:
all mapping